### ISSUE TITLE
Autocomplete for files doesn't work

### ISSUE THREAD

**Issue by armak** [cmderdev/cmder#826]
Autocompletion by pressing tab doesn't work in the latest Cmder I downloaded from the official website. This happens with the default settings (including a complete Reset...) and I couldn't find any setting that could fix this. Autocomplete works fine on directories and commands though.

**Comment by okoetter** [comment]
I have the same problem. E.g. when I do a `git add filename` I get no autocompletion for filename. Is there a workaround until this gets fixed?

cmder 1.2.9
ConEmu 150913 [32]
Clink v0.4.7

**Comment by MartiUK** [comment]
Does this build resolve your issues?

https://ci.appveyor.com/project/MartiUK/cmder/build/1.0.186-master/artifacts

**Comment by armak** [comment]
It doesn't, same behavior exhibited.

**Comment by kkthx** [comment]
Same here with git. Video: http://sendvid.com/efqrkp9g
I tried to install newest clink and conemu but this doesn't resolve this bug so problem is with cmder

**Comment by JanSchulz** [comment]
A workaround could be `git add -- file`.

I don't see this problem, `git add file` complets here. Could it be that you have old versions of clink and so on in the cmder dir?

**Comment by armak** [comment]
I did some experimentation earlier. The issue is "solved" if you remove clink entirely from vendor/clink/, but this causes various issues with text output:

![cmder1](https://cloud.githubusercontent.com/assets/4991927/13470915/f8c128c6-e0b6-11e5-9726-361c0564ae1c.png)

Replacing clink with other versions either does nothing or causes equivalent issues. However, replacing the clink component from my current, much older version of Cmder with the clink version supplied with the most recent Cmder doesn't affect the behavior and file completion still works normally in the old Cmder.

I'm also using Windows 7, if that makes a difference.

**Comment by rakete** [comment]
I just had the same problem with 1.2.9 and clink 0.4.7.

I tested with 1.3.0-pre and with that version auto-completion works correctly.

**Comment by okoetter** [comment]
I downloaded 1.3.0-pre with ConEmu 160328 and it also works for me now.

**Comment by armak** [comment]
Still doesn't work for me on 1.3.0-pre. Completion currently only works for directories, .exe files and for some reason .js files. Possibly some others too but those are the only ones I noticed to work.

**Comment by mortona42** [comment]
I had the problem (no tab completion for files after [git add ]).  Upgrading to 1.3.0 fixed it for me,  I think I had 1.3.0-pre before then.

**Comment by CIPop** [comment]
Latest 1.3.2 still has this issue: no autocompletion for files within the current directory.

**Comment by okitu** [comment]
This here solved it for me:
https://github.com/cmderdev/cmder/issues/508#issuecomment-275409806

**Comment by armak** [comment]
Yes, the setting mentioned okitu seems to finally resolve this for good!

**Comment by drspangle** [comment]
I'm using Cmder latest with ConEmu 161206 x64 stable, and I can confirm that okitu's fix doesn't work. With no text on the prompt, the tab autocomplete does now correctly list the folder contents of the current working directory, but when anything is typed in matching only occurs with executables found in the PATH, not the current working directory, and not with anything other than executables.

To clarify what I did, re: #508, I modified the existing line setting `exec_match_style = 2` to `exec_match_style = -1`

**Comment by ihavenonickname** [comment]
#508 didn't do the trick for me, same as @drspangle 

I have a `scripts` folder in PATH with some powershell scripts. I execute them a lot and it would be *much* easier with tab completion.

**Comment by OlegGorenko** [comment]
I deleted the file **<%CMDER-ROOT%>\config\settings** and restarted cmder.

Now I have auto-completion.

**Comment by Dimon4eg** [comment]
For me also https://github.com/cmderdev/cmder/issues/508 is not solution.

I tried `exec_match_style = -1` but it introduces another problem - autocomplete doesn't display executable from PATH:

1) exec_match_style = 2
executable from PATH - OK
scripts from directory - FAIL

2) exec_match_style = -1
executable from PATH - FAIL
scripts from directory - OK

How to make it work for both cases?

**Comment by daxgames** [comment]
Seems to work in the current version.  Closing.

**Comment by daxgames** [comment]
What you are expecting is wrong.  Tab completion works as it would on a Linux system, the current folder is not in the path and must be specified using `.\` or `./`.  Try:

```
.\bu[tab]
```

**Comment by Dimon4eg** [comment]
`.\` or `./` doesn't work

**Comment by daxgames** [comment]
It does for me.  What version you running?


![cmder_tab_complete](https://user-images.githubusercontent.com/7318053/49347632-43417980-f66e-11e8-90cd-e6ffd88557c7.gif)

**Comment by Dimon4eg** [comment]
I specified version above.
Can you try steps as I described?

**Comment by daxgames** [comment]
I did and you are correct `build.py` does not auto complete but this is not a Cmder issue.  You are trying to auto complete the first word of a line therefore a command to be run so the file must be executable in `cmd.exe`.

On Windows in `cmd.exe` sessions executeable file extensions are configured by the `PATHEXT` environment variable.  Mine is currently set to:

```
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
```

If I add `.py` to it autocomplete works.

**Comment by Dimon4eg** [comment]
Cool! Now it works as expected.
Thank you @daxgames !

**Comment by AlanTrudor** [comment]
Hey I am facing the same issue with the latest release of v1.3.18.1106, and so far none of the above solutions have worked for me. Did someone manage to resolve it?

**Comment by chrisant996** [comment]
Instead of adding a reply to 3 different closed issues (#826, #508, #218), please open a new issue, and please include the output from `clink set` to share what your settings are.
